### PR TITLE
Enhance flexibility by passing callbacks as method argument

### PR DESCRIPTION
**Pull request by michal2409**
# What does this PR do ?

This PR enhances the flexibility of CustomProgressBar by modifying the create_trainer method to accept callbacks as a method argument, replacing its previous fixed implementation.

**Collection**: NLP

# Changelog 
Moved callbacks to method arguement.

# Usage
N/A

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] Refactor
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by janekl** [comment]
@titu1994 could you also please help to review this? Short one, focusing on UX

File: nemo/collections/nlp/parts/megatron_trainer_builder.py [hunk 1]
```diff
@@ -95,10 +95,10 @@ def _plugins(self) -> list:
 
         return plugins
 
-    def create_trainer(self) -> Trainer:
+    def create_trainer(self, callbacks=(CustomProgressBar(),)) -> Trainer:
```
Comment (titu1994) [inline nemo/collections/nlp/parts/megatron_trainer_builder.py:98]: Don't instantiate objects in function signature. Default to None, check if None inside the function and the. Instantiate the value you need. Only primitive values are safe in function signature. https://leimao.github.io/blog/Python-Default-Argument-Mutable-Object/

Comment (michal2409) [reply nemo/collections/nlp/parts/megatron_trainer_builder.py]: Fixed

**Comment by janekl** [comment]
@titu1994 could you have another look?

**Pull request merged**